### PR TITLE
Implement core metrics

### PR DESCRIPTION
**Pull request by xplosunn**

### CODE REVIEW

File: core/src/main/scala/com/codacy/analysis/core/tools/MetricsTool.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  private val availableTools = PluginHelper.dockerMetricsPlugins
+
+  def fromLanguages(languages: Set[Language]): Set[MetricsTool] = {
+    languages.flatMap { lang =>
+      val collectedTools = availableTools.collect {
+        case tool if tool.languages.contains(lang) =>
+          new MetricsTool(tool, Some(lang))
+      }
+      if (collectedTools.isEmpty) {
+        logger.info(s"No duplication tools found for language ${lang.name}")
```
Comment (bmbferreira) [inline core/src/main/scala/com/codacy/analysis/core/tools/MetricsTool.scala]: "No metrics tools found..."

File: cli/src/main/scala/com/codacy/analysis/cli/formatter/Text.scala [hunk 2]
```diff
@@ -36,6 +36,31 @@ private[formatter] class Text(val stream: PrintStream) extends Formatter {
...
+      fileMetrics.cloc.map(prettyNamedValue("commented lines of code", _)),
+      fileMetrics.nrMethods.map(prettyNamedValue("number of methods", _)),
+      fileMetrics.nrClasses.map(prettyNamedValue("number of classes", _)))
+
+    val fileMetricsValues = fileMetricsList.collect {
+      case Some(namedValue) => namedValue
+    }
+
+    if (fileMetricsValues.isEmpty) {
+      s"No file metrics found on file ${fileMetrics.filename}"
```
Comment (rtfpessoa) [inline cli/src/main/scala/com/codacy/analysis/cli/formatter/Text.scala]: `No file metrics found on file ...` -> `No metrics found on file ...`

File: cli/src/main/scala/com/codacy/analysis/cli/formatter/Text.scala [hunk 3]
```diff
@@ -36,6 +36,31 @@ private[formatter] class Text(val stream: PrintStream) extends Formatter {
...
+    } else {
+      s"Found ${fileMetricsValues.mkString(",")} on file ${fileMetrics.filename}"
```
Comment (rtfpessoa) [inline cli/src/main/scala/com/codacy/analysis/cli/formatter/Text.scala]: Not sure how this looks but maybe multi-line would be nice.

File: project/Dependencies.scala [hunk 4]
```diff
@@ -27,7 +27,7 @@ object Dependencies {
   lazy val caseApp = "com.github.alexarchambault" %% "case-app" % "1.2.0"
 
   lazy val codacyPlugins =
-    Seq("com.codacy" %% "codacy-plugins-api" % "3.0.96", "codacy" %% "codacy-plugins" % "5.0.92")
+    Seq("com.codacy" %% "codacy-plugins-api" % "3.0.96", "codacy" %% "codacy-plugins" % "metrics-languages-SNAPSHOT")
```
Comment (rtfpessoa) [inline project/Dependencies.scala]: Update to `5.0.117`

**Pull request merged**